WebAssembly | Memory - homogenise descriptions, and add note about SAB (#27021)
mdn/content: files/en-us/webassembly/javascript_interface/memory/grow/index.md

// 131072 = 64KiB * 2
```

> For a shared `Memory` instance, the initial `buffer` (which would be a `SharedArrayBuffer` in such case) will not become detached, but rather its length will not update. It will still be a view onto the same memory span. Accesses to the `buffer` property after growing will yield a larger `SharedArrayBuffer` which may access a larger span of memory than the buffer from before growing the `Memory`.

Reviewer: Could we reword this:

>It will still be a view onto the same memory span.

I'm not sure what this means.

Author: > Could we reword this:
> 
> > It will still be a view onto the same memory span.
> 
> I'm not sure what this means.

I mean to say that it refers to the same memory. They are overlapping memory address ranges.

> For a shared `Memory` instance, the initial `buffer` (which would be a `SharedArrayBuffer` in such case) will not become detached, but rather its length will not update. It will still be a view onto the same memory span. Accesses to the `buffer` property after growing will yield a larger `SharedArrayBuffer` which may access a larger span of memory than the buffer from before growing the `Memory`.

Reviewer: Adding some links and rewording for clarity. Could you check this is accurate?

```suggestion
For a shared `Memory` instance, the initial `buffer` (which would be a [`SharedArrayBuffer`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer) in such case) will not become detached. The shared `buffer` length will not update and the same memory span will still be accessible. Access to the `buffer` property after growing will yield a larger `SharedArrayBuffer` which may access a larger span of memory than was allocated before calling `grow()`.
```

> For a shared `Memory` instance, the initial `buffer` (which would be a [`SharedArrayBuffer`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer) in such case) will not become detached, but rather its length will not be updated. Accesses to the `buffer` property after growing will yield a larger `SharedArrayBuffer` which may access a larger span of memory than the buffer from before growing the `Memory`. Every `SharedArrayBuffer` from the `buffer` property will all refer to the start of the same memory address range, and thus manipulate the same data.

Reviewer: That's one for @Phosra - is there a typo in "rather its length will not be updated"?

> For a shared `Memory` instance, the initial `buffer` (which would be a [`SharedArrayBuffer`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer) in such case) will not become detached, but rather its length will not be updated. Accesses to the `buffer` property after growing will yield a larger `SharedArrayBuffer` which may access a larger span of memory than the buffer from before growing the `Memory`. Every `SharedArrayBuffer` from the `buffer` property will all refer to the start of the same memory address range, and thus manipulate the same data.

Author: "will not be updated" the length is constant from construction.

Hmm, should this change have been accompanied by a code block to also explain it for others, as I had done here explaining it?

> For a shared `Memory` instance, the initial `buffer` (which would be a [`SharedArrayBuffer`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer) in such case) will not become detached, but rather its length will not be updated. Accesses to the `buffer` property after growing will yield a larger `SharedArrayBuffer` which may access a larger span of memory than the buffer from before growing the `Memory`. Every `SharedArrayBuffer` from the `buffer` property will all refer to the start of the same memory address range, and thus manipulate the same data.

Reviewer: >should this change have been accompanied by a code block to also explain it for others, as I had done here explaining it

That's definitely a good follow-up. Would you like to take the example from this discussion and propose changes in a new PR?

> For a shared `Memory` instance, the initial `buffer` (which would be a [`SharedArrayBuffer`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer) in such case) will not become detached, but rather its length will not be updated. Accesses to the `buffer` property after growing will yield a larger `SharedArrayBuffer` which may access a larger span of memory than the buffer from before growing the `Memory`. Every `SharedArrayBuffer` from the `buffer` property will all refer to the start of the same memory address range, and thus manipulate the same data.

Author: Yes that was my plan. I'll be on it in a few hours.

> For a shared `Memory` instance, the initial `buffer` (which would be a [`SharedArrayBuffer`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer) in such case) will not become detached, but rather its length will not be updated. Accesses to the `buffer` property after growing will yield a larger `SharedArrayBuffer` which may access a larger span of memory than the buffer from before growing the `Memory`. Every `SharedArrayBuffer` from the `buffer` property will all refer to the start of the same memory address range, and thus manipulate the same data.

Reviewer: Sounds great, thank you. I'll resolve this conversation for now.
